src/code/z_actor.c

```diff
@@ -471,237 +484,241 @@ void Target_SetColors(TargetContext* targetCtx, Actor* actor, s32 type, PlayStat
 
-            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, color->inner.r, color->inner.g, color->inner.b, 255);
-            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_MODELVIEW | G_MTX_LOAD);
-            gSPDisplayList(POLY_XLU_DISP++, gZTargetArrowDL);
-        }
+        Matrix_Translate(actor->focus.pos.x, actor->focus.pos.y + (actor->targetArrowOffset * actor->scale.y) + 17.0f,
+                         actor->focus.pos.z, MTXMODE_NEW);
+        Matrix_RotateYS(play->gameplayFrames * 0xBB8, MTXMODE_APPLY);
+        Matrix_Scale((iREG(27) + 35) / 1000.0f, (iREG(28) + 60) / 1000.0f, (iREG(29) + 50) / 1000.0f, MTXMODE_APPLY);
 
-        CLOSE_DISPS(play->state.gfxCtx);
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, color->inner.r, color->inner.g, color->inner.b, 255);
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_MODELVIEW | G_MTX_LOAD);
+        gSPDisplayList(POLY_XLU_DISP++, gZTargetArrowDL);
     }
+
+    CLOSE_DISPS(play->state.gfxCtx);
 }
 
 // OoT: func_8002C7BC
-void func_800B5814(TargetContext* targetCtx, Player* player, Actor* actor, GameState* gameState) {
-    PlayState* play = (PlayState*)gameState;
-    Actor* sp68 = NULL;
+void Target_Update(TargetContext* targetCtx, Player* player, Actor* targetedActor, PlayState* play) {
+    s32 pad;
+    Actor* actor = NULL;
     s32 category;
     Vec3f projectedPos;
     f32 invW;
 
+    // If no actor is currently targeted, then try to find a targetable actor
     if ((player->targetedActor != NULL) && (player->unk_AE3[player->unk_ADE] == 2)) {
-        targetCtx->unk_94 = NULL;
+        targetCtx->arrowPointedActor = NULL;
     } else {
-        func_800BB8EC(gameState, &play->actorCtx, &sp68, &D_801ED920, player);
-        targetCtx->unk_94 = sp68;
+        Target_GetTargetActor(play, &play->actorCtx, &actor, &D_801ED920, player);
+        targetCtx->arrowPointedActor = actor;
     }
 
-    if (targetCtx->unk8C != NULL) {
-        sp68 = targetCtx->unk8C;
-        targetCtx->unk8C = NULL;
-    } else if (actor != NULL) {
-        sp68 = actor;
+    if (targetCtx->nextTargetableOption != NULL) {
+        actor = targetCtx->nextTargetableOption;
+        targetCtx->nextTargetableOption = NULL;
+    } else if (targetedActor != NULL) {
+        actor = targetedActor;
     }
 
-    if (sp68 != NULL) {
-        category = sp68->category;
+    if (actor != NULL) {
+        category = actor->category;
     } else {
         category = player->actor.category;
     }
 
-    if ((sp68 != targetCtx->arrowPointedActor) || (category != targetCtx->unk4A)) {
-        targetCtx->arrowPointedActor = sp68;
-        targetCtx->unk4A = category;
-        targetCtx->unk40 = 1.0f;
+    if ((actor != targetCtx->targetableOption) || (category != targetCtx->targetableOptionCategory)) {
+        targetCtx->targetableOption = actor;
+        targetCtx->targetableOptionCategory = category;
+        targetCtx->fairyMoveProgressFactor = 1.0f;
```

Comment:
From this block and the next block which updates `fairyHintPos`/`fairyPos`, I think `targetableOption` should be `fairyActor` and `targetableOptionCategory` should be `fairyActorCategory`. This seems to be the actor that navi/tatl will always be after, since `fairyMoveProgressFactor` is associated with this actor.

If you're not z-targeting an actor but an actor is targetable, navi will fly over to that actor, so I can see where `targetableOption` comes from.

However, if you're z-targeting an actor and you have the option to switch targeting to another actor, `targetableOption` will remain on the current actor while `arrowPointedActor` will move to the next actor to show where you would switch. So `targetableOption` doesn't really make sense in that context